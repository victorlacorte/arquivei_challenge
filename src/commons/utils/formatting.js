function toBRL(number) {
  return number.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

export {
  toBRL,
};
