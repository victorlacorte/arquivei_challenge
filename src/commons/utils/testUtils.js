/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { ThemeProvider } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount, shallow } from 'enzyme';

import defaultTheme from 'commons/styles/theme';

function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

export {
  mountWithTheme,
  shallowWithTheme,
};
