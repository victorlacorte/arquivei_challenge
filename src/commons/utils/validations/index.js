export { default as cnpjValidations } from './cnpj';
export { default as consultKeysValidations } from './consultKeys';
export { default as creditCardValidations } from './creditCard';
export { default as formValidations } from './formValidations';
