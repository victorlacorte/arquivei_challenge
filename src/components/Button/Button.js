import { memo } from 'react';
import styled, { css } from 'styled-components';

// TODO add proptype validation? Also needs to be tested
const StyledButton = styled.button.attrs((props) => ({
  type: props.type || 'button',
  disabled: props.disabled || false,
  variant: props.variant || 'contained',
  color: props.color || 'primary',
}))`
  box-sizing: border-box;
  min-height: 40px;
  padding: 0px 32px;
  border: none;
  border-radius: 24px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  outline: none;
  color: white;

  ${({ theme }) => css`
    ${theme.typography.button};
  `}

  ${({ disabled, variant, color }) => !disabled && variant === 'contained' && css`
    background-color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].dark};
    }
  `}

  ${({ disabled, variant, color }) => !disabled && variant === 'outlined' && css`
    background-color: white;
    border: 2px solid ${({ theme }) => theme.palette[color].main};
    color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].light};
    }
  `}

  ${({ disabled }) => disabled && css`
    pointer-events: none;
    cursor: default;
    background-color: ${({ theme }) => theme.palette.grey400.main};
  `}
`;

export default memo(StyledButton);
