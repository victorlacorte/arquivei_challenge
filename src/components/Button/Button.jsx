import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

/* eslint-disable react/button-has-type */
function Button(props) {
  const {
    className,
    type,
    disabled,
    children,
  } = props;

  return (
    <button
      className={className}
      type={type}
      disabled={disabled}
    >
      {children}
    </button>
  );
}

Button.defaultProps = {
  type: 'button',

  disabled: false,
};

Button.propTypes = {
  className: PropTypes.string.isRequired,

  type: PropTypes.oneOf(['button', 'submit']),

  disabled: PropTypes.bool,

  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

const StyledButton = styled(Button).attrs(({ variant, color }) => ({
  variant: variant || 'contained',
  color: color || 'primary',
}))`
  box-sizing: border-box;
  min-height: 40px;
  padding: 0px 32px;
  border: none;
  border-radius: 24px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  outline: none;
  color: white;

  ${({ theme }) => css`
    ${theme.typography.button};
  `}

  ${({ variant, color }) => variant === 'contained' && css`
    background-color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].dark};
    }
  `}

  ${({ variant, color }) => variant === 'outlined' && css`
    background-color: white;
    border: 2px solid ${({ theme }) => theme.palette[color].main};
    color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].light};
    }
  `}

  ${({ disabled }) => disabled && css`
    pointer-events: none;
    cursor: default;
    background-color: ${({ theme }) => theme.palette.grey[400]};
  `}
`;

export default StyledButton;
