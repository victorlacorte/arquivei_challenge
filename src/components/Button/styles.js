import { memo } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const StyledButton = styled.button.attrs((props) => ({
  type: props.type || 'button',
  disabled: props.disabled || false,
  variant: props.variant || 'contained',
  color: props.color || 'primary',
}))`
  box-sizing: border-box;
  min-height: 40px;
  padding: 0px 32px;
  border: none;
  border-radius: 24px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  outline: none;
  color: white;

  ${({ theme }) => theme.breakpoints.down('sm')`
    width: 75%;
  `}

  ${({ theme }) => css`
    ${theme.typography.button};
  `}

  ${({ disabled, variant, color }) => !disabled && variant === 'contained' && css`
    background-color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].dark};
    }
  `}

  ${({ disabled, variant, color }) => !disabled && variant === 'outlined' && css`
    background-color: white;
    border: 2px solid ${({ theme }) => theme.palette[color].main};
    color: ${({ theme }) => theme.palette[color].main};

    :hover {
      background-color: ${({ theme }) => theme.palette[color].light};
    }
  `}

  ${({ disabled }) => disabled && css`
    pointer-events: none;
    cursor: default;
    background-color: ${({ theme }) => theme.palette.grey400.main};
  `}
`;

StyledButton.propTypes = {
  type: PropTypes.oneOf(['button', 'submit']),

  disabled: PropTypes.bool,

  variant: PropTypes.oneOf(['contained', 'outlined']),

  color: PropTypes.oneOf(['primary', 'success', 'error', 'alert']),
};

export default memo(StyledButton);
