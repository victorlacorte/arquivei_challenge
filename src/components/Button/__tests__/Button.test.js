import React from 'react';

import Button from 'components/Button';
import { shallowWithTheme } from 'commons/utils/testUtils';

describe('<Button />', () => {
  it('creates a Button component', () => {
    const wrapper = shallowWithTheme(<Button />);

    expect(wrapper).toMatchSnapshot();
  });

  it('creates a disabled Button component', () => {
    const wrapper = shallowWithTheme(<Button disabled />);

    expect(wrapper.props().disabled).toBe(true);
  });
});
