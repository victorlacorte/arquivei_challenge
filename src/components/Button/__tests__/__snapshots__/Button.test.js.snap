// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> creates a Button component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "styles__StyledButton-sc-1u6a504-0",
        "isStatic": false,
        "rules": Array [
          "box-sizing:border-box;min-height:40px;padding:0px 32px;border:none;border-radius:24px;display:inline-flex;justify-content:center;align-items:center;outline:none;color:white;",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
        ],
      },
      "displayName": "styles__StyledButton",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "color": [Function],
        "disabled": [Function],
        "type": [Function],
        "variant": [Function],
      },
      "render": [Function],
      "styledComponentId": "styles__StyledButton-sc-1u6a504-0",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
